app/src/main/java/app/revanced/manager/ui/screen/PatcherScreen.kt

```diff
@@ -0,0 +1,433 @@
+                .clickable { expanded = !expanded }
+                .background(categoryColor)
+        ) {
+            Row(
+                verticalAlignment = Alignment.CenterVertically,
+                horizontalArrangement = Arrangement.spacedBy(12.dp),
+                modifier = Modifier.padding(16.dp)
+            ) {
+                val state = remember(steps) {
+                    when {
+                        steps.all { it.state == State.COMPLETED } -> State.COMPLETED
+                        steps.any { it.state == State.FAILED } -> State.FAILED
+                        steps.any { it.state == State.RUNNING } -> State.RUNNING
+                        else -> State.WAITING
+                    }
+                }
+
+                StepIcon(
+                    state = state,
+                    size = 24.dp
+                )
+
+                Text(
+                    stringResource(category.displayName),
+                    style = MaterialTheme.typography.titleMedium
+                )
+
+                Spacer(modifier = Modifier.weight(1f))
+
+                Text(
+                    "${steps.count { it.state == State.COMPLETED }}/${steps.size}",
+                    style = MaterialTheme.typography.labelSmall
+                )
+
+                ArrowButton(modifier = Modifier.size(24.dp), expanded = expanded, onClick = null)
+            }
+        }
+
+        AnimatedVisibility(visible = expanded) {
+            Column(
+                modifier = Modifier.fillMaxWidth()
+            ) {
+                if (category != StepCategory.PATCHING) {
+                    steps.forEach { step ->
+                        val downloadProgress = step.downloadProgress?.collectAsStateWithLifecycle()
+
+                        SubStep(
+                            name = step.name,
+                            state = step.state,
+                            message = step.message,
+                            downloadProgress = downloadProgress?.value
+                        )
+                    }
+                } else {
+                    val currentPatch = remember(steps) {
+                        steps.find { it.state == State.RUNNING }?.name
+                    }
+                    val failedPatch = remember(steps) {
+                        steps.find { it.state == State.FAILED }?.name
+                    }
+
+                    val name = remember(steps) {
+                        when {
+                            steps.any { it.state == State.RUNNING } -> context.getString(R.string.applying_patch, currentPatch)
+                            steps.any { it.state == State.FAILED } -> context.getString(R.string.failed_to_apply_patch, failedPatch)
+                            steps.all { it.state == State.WAITING } -> context.getString(R.string.apply_patches)
+                            else -> context.resources.getQuantityString(R.plurals.applied_patches, steps.size, steps.size)
```

Comment:
In my opinion, this looks rather bad, each patch should be given its individual substep, this substep currently gives the same information as the Step header itself

Reply:
You should discuss this with Palm